### PR TITLE
Comply with REUSE 3.0

### PR DESCRIPTION
**Pull request by fosslinux**
REUSE is a specification on how repositories should ensure licensing is done properly. Since our licensing is not very simple, I have decided we should use this.

However, I need existing contributors to be clear about the following:

- code changes are licensed under the GNU GPL v3.0 (applicable to @stikonas @bittorf @pder)
- documentation changes are licensed under Creative Commons BY-SA 4.0 (applicable to @stikonas @pder)
- patches are licensed under the license of the project of which they are patching (applicable to @stikonas @pder)

Please ensure that you have been attributed everywhere you should be; if you have time, please also check that I haven't added people (including myself!) where they shouldn't be.

Approving this pull request with a note of something like "I agree to this licensing" should cover any potential issues.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,3 +1,8 @@
+# SPDX-FileCopyrightText: 2021 Andrius Štikonas <andrius@stikonas.eu>
+# SPDX-FileCopyrightText: 2021 fosslinux <fosslinux@aussies.space>
+#
+# SPDX-License-Identifier: GPL-3.0-or-later
```
Comment (stikonas) [inline .gitignore:4]: I would use something simpler for .gitignore... It's just a list of files, so maybe more permissive?

File: .gitmodules [hunk 2]
```diff
@@ -1,3 +1,8 @@
+# SPDX-FileCopyrightText: 2021 Andrius Štikonas <andrius@stikonas.eu>
+# SPDX-FileCopyrightText: 2021 fosslinux <fosslinux@aussies.space>
+#
+# SPDX-License-Identifier: GPL-3.0-or-later
```
Comment (stikonas) [inline .gitmodules:4]: If we change .gitignore, maybe this one too...

File: rootfs.sh [hunk 3]
```diff
@@ -176,4 +183,4 @@ esac
 cd ../..
 
 # Cleanup
-sudo umount sysa/tmp
+#sudo umount sysa/tmp
```
Comment (stikonas) [inline rootfs.sh:186]: intentional?

File: sysa/blynn-compiler/blynn-compiler.kaem [hunk 4]
```diff
@@ -1,5 +1,9 @@
 #!/bin/sh
 
+# SPDX-FileCopyrightText: 2021 fosslinux <fosslinux@aussies.space>
+#
+# SPDX-License-Identifier: GPL-3.0-or-later
+
```
Comment (stikonas) [inline sysa/blynn-compiler/blynn-compiler.kaem:6]: This is basically exact copy of `go.sh` from blynn-compiler. We should probably add attributions from there...
Especially for @pder and @oriansj

File: sysa/mes/files/mescc.scm [hunk 5]
```diff
@@ -1,20 +1,6 @@
...
-;;; your option) any later version.
-;;;
-;;; GNU Mes is distributed in the hope that it will be useful, but
-;;; WITHOUT ANY WARRANTY; without even the implied warranty of
-;;; MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
-;;; GNU General Public License for more details.
-;;;
-;;; You should have received a copy of the GNU General Public License
-;;; along with GNU Mes.  If not, see <http://www.gnu.org/licenses/>.
+; Copyright © 2016,2017,2018,2019 Jan (janneke) Nieuwenhuizen <janneke@gnu.org>
```
Comment (stikonas) [inline sysa/mes/files/mescc.scm:1]: `SPDX-FileCopyrightText` ?

File: sysa/mescc-tools-extra/src/chmod.c [hunk 6]
```diff
@@ -1,19 +1,7 @@
...
- *
- * mescc-tools is distributed in the hope that it will be useful,
- * but WITHOUT ANY WARRANTY; without even the implied warranty of
- * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
- * GNU General Public License for more details.
- *
- * You should have received a copy of the GNU General Public License
- * along with mescc-tools.  If not, see <http://www.gnu.org/licenses/>.
- */
+// Copyright (C) 2020 fosslinux
```
Comment (stikonas) [inline sysa/mescc-tools-extra/src/chmod.c:1]: Remove this line. There is SPDX-FileCopyrightText on the next line

File: sysa/mescc-tools-extra/src/cp.c [hunk 7]
```diff
@@ -1,19 +1,7 @@
...
- *
- * mescc-tools is distributed in the hope that it will be useful,
- * but WITHOUT ANY WARRANTY; without even the implied warranty of
- * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
- * GNU General Public License for more details.
- *
- * You should have received a copy of the GNU General Public License
- * along with mescc-tools.  If not, see <http://www.gnu.org/licenses/>.
- */
+// Copyright (C) 2020 fosslinux
```
Comment (stikonas) [inline sysa/mescc-tools-extra/src/cp.c:1]: same here

File: sysa/mescc-tools-extra/src/functions/file_print.c [hunk 8]
```diff
@@ -1,19 +1,7 @@
...
- *
- * mescc-tools is distributed in the hope that it will be useful,
- * but WITHOUT ANY WARRANTY; without even the implied warranty of
- * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
- * GNU General Public License for more details.
- *
- * You should have received a copy of the GNU General Public License
- * along with mescc-tools.  If not, see <http://www.gnu.org/licenses/>.
- */
+// Copyright (C) 2016 Jeremiah Orians
```
Comment (stikonas) [inline sysa/mescc-tools-extra/src/functions/file_print.c:1]: SPDX-FileCopyrightText

File: sysa/mescc-tools-extra/src/functions/in_set.c [hunk 9]
```diff
@@ -1,20 +1,7 @@
...
- *
- * mescc-tools is distributed in the hope that it will be useful,
- * but WITHOUT ANY WARRANTY; without even the implied warranty of
- * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
- * GNU General Public License for more details.
- *
- * You should have received a copy of the GNU General Public License
- * along with mescc-tools.  If not, see <http://www.gnu.org/licenses/>.
- */
+// Copyright (C) 2016 Jeremiah Orians
```
Comment (stikonas) [inline sysa/mescc-tools-extra/src/functions/in_set.c:1]: SPDX-FileCopyrightText

File: sysa/mescc-tools-extra/src/functions/match.c [hunk 10]
```diff
@@ -1,19 +1,6 @@
...
- *
- * mescc-tools is distributed in the hope that it will be useful,
- * but WITHOUT ANY WARRANTY; without even the implied warranty of
- * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
- * GNU General Public License for more details.
- *
- * You should have received a copy of the GNU General Public License
- * along with mescc-tools.  If not, see <http://www.gnu.org/licenses/>.
- */
+// Copyright (C) 2016 Jeremiah Orians
```
Comment (stikonas) [inline sysa/mescc-tools-extra/src/functions/match.c:1]: SPDX-FileCopyrightText

File: sysa/mescc-tools-extra/src/functions/numerate_number.c [hunk 11]
```diff
@@ -1,19 +1,6 @@
...
- *
- * mescc-tools is distributed in the hope that it will be useful,
- * but WITHOUT ANY WARRANTY; without even the implied warranty of
- * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
- * GNU General Public License for more details.
- *
- * You should have received a copy of the GNU General Public License
- * along with mescc-tools.  If not, see <http://www.gnu.org/licenses/>.
- */
+// Copyright (C) 2016 Jeremiah Orians
```
Comment (stikonas) [inline sysa/mescc-tools-extra/src/functions/numerate_number.c:1]: SPDX-FileCopyrightText

File: sysa/mescc-tools-extra/src/functions/require.c [hunk 12]
```diff
@@ -1,20 +1,7 @@
...
- *
- * mescc-tools is distributed in the hope that it will be useful,
- * but WITHOUT ANY WARRANTY; without even the implied warranty of
- * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
- * GNU General Public License for more details.
- *
- * You should have received a copy of the GNU General Public License
- * along with mescc-tools.  If not, see <http://www.gnu.org/licenses/>.
- */
+// Copyright (C) 2016 Jeremiah Orians
```
Comment (stikonas) [inline sysa/mescc-tools-extra/src/functions/require.c:1]: SPDX-FileCopyrightText

File: sysa/mescc-tools-extra/src/functions/string.c [hunk 13]
```diff
@@ -1,19 +1,6 @@
...
- *
- * mescc-tools is distributed in the hope that it will be useful,
- * but WITHOUT ANY WARRANTY; without even the implied warranty of
- * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
- * GNU General Public License for more details.
- *
- * You should have received a copy of the GNU General Public License
- * along with mescc-tools.  If not, see <http://www.gnu.org/licenses/>.
- */
+// Copyright (C) 2016 Jeremiah Orians
```
Comment (stikonas) [inline sysa/mescc-tools-extra/src/functions/string.c:1]: SPDX-FileCopyrightText

**Review by stikonas — CHANGES_REQUESTED** [review]
I agree to licensing changes that were described in the comment:

- code changes are licensed under the GNU GPL v3.0
- documentation changes are licensed under Creative Commons BY-SA 4.0
- patches are licensed under the license of the project of which they are patching

**Comment by bittorf** [comment]
I agree to this licensing.

**Comment by pder** [comment]
I agree to this licensing.

File: sysa/tcc-0.9.27/patches/mes-libc-qsort.patch [hunk 14]
```diff
@@ -1,16 +1,12 @@
...
 --- lib/stdlib/qsort.c
 +++ lib/stdlib/qsort.c
-@@ -1,6 +1,7 @@
- /* -*-comment-start: "//";comment-end:""-*-
-  * GNU Mes --- Maxwell Equations of Software
-  * Copyright © 2017,2018 Jan (janneke) Nieuwenhuizen <janneke@gnu.org>
-+ * Copyright © 2021 Paul Dersey <pdersey@gmail.com>
-  *
-  * This file is part of GNU Mes.
-  *
```
Comment (stikonas) [inline sysa/tcc-0.9.27/patches/mes-libc-qsort.patch:13]: Is this removal intentional?

Comment (fosslinux) [reply sysa/tcc-0.9.27/patches/mes-libc-qsort.patch:13]: Yeah, we don't do this in any other patch and appropriate attribution has been given through the use of SPDX-Copyright header.

**Pull request merged**